File: 108. Convert Sorted Array to Binary Search Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        if (nums.length == 0) {
            return null;
        }
        TreeNode output = bst(nums, 0, nums.length - 1);
        
        return output;
    }
    
    public TreeNode bst(int[] nums, int curr, int end) {
        int mid = (end + curr)/2;
        
        if (curr > end) {
            return null;
        }
        TreeNode root = new TreeNode(nums[mid]);
        root.left = bst(nums, curr, mid - 1);
        root.right = bst(nums, mid + 1, end);
        
        return root;
    }
}

File: 108. Convert Sorted Array to Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sortedArrayToBST(self, nums):
        """
        :type nums: List[int]
        :rtype: TreeNode
        """
        def bst(inpt, start, end):
            if start > end:
                return None
            
            middle = (start + end) // 2
            node = TreeNode(inpt[middle])
            node.left = bst(inpt, start, middle - 1)
            node.right = bst(inpt, middle + 1, end)
            
            return node
            
        if not nums:
            return None
        
        return bst(nums, 0, len(nums) - 1)
